lib/util/historical-buffer-manager.ts

```diff
@@ -0,0 +1,95 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import interfaces = require('../interface');
+
+// Internal class that holds the buffer array, a counter for overflow items and maxBufferLength
+class Buffer<T> {
+    data: T[];
```

Comment:
I am not sure if I want to mark the `data` field as private, especially this is an internal class. I can definitely add all access control over this class, just not sure if it is best practice to do it over internal class.